api/src/main/java/org/openmrs/module/idgen/IdgenModuleActivator.java

```diff
@@ -30,6 +35,19 @@ public class IdgenModuleActivator extends BaseModuleActivator implements DaemonT
 
 	@Override
 	public void started() {
+		IdentifierSourceService service = Context.getService(IdentifierSourceService.class);
+		service.registerProcessor(
+				SequentialIdentifierGenerator.class,
+				Context.getRegisteredComponents(SequentialIdentifierGeneratorProcessor.class).get(0)
+		);
+		service.registerProcessor(
+				RemoteIdentifierSource.class,
+				Context.getRegisteredComponents(RemoteIdentifierSourceProcessor.class).get(0)
+		);
+		service.registerProcessor(
+				IdentifierPool.class,
+				Context.getRegisteredComponents(IdentifierPoolProcessor.class).get(0)
+		);
```

Comment:
The way this was originally set up, the SequentialIdentifierGenerator bean has a reference to the identifierSourceService bean, and the BaseIdentifierSource implementation has a reference to the SequentialIdentifierGenerator bean, so there was a circular dependency between these 2 beans.  I don't know what changed, but this does not seem to be something that is no longer ok with the current setup.

So I removed the wiring of processors into the service from the application config xml, added a Component annotation onto the processors to register them as beans, and then am wiring them into the service in the activator.

Ultimately, this should result in the same functionality.